src/main/java/org/betonquest/betonquest/compatibility/traincarts/objectives/TrainCartsRideObjective.java

```diff
@@ -0,0 +1,175 @@
+package org.betonquest.betonquest.compatibility.traincarts.objectives;
+
+import com.bergerkiller.bukkit.tc.events.seat.MemberSeatEnterEvent;
+import com.bergerkiller.bukkit.tc.events.seat.MemberSeatExitEvent;
+import org.betonquest.betonquest.BetonQuest;
+import org.betonquest.betonquest.Instruction;
+import org.betonquest.betonquest.api.CountingObjective;
+import org.betonquest.betonquest.api.config.quest.QuestPackage;
+import org.betonquest.betonquest.api.logger.BetonQuestLogger;
+import org.betonquest.betonquest.api.profiles.OnlineProfile;
+import org.betonquest.betonquest.exceptions.InstructionParseException;
+import org.betonquest.betonquest.exceptions.QuestRuntimeException;
+import org.betonquest.betonquest.instruction.variable.VariableNumber;
+import org.betonquest.betonquest.instruction.variable.VariableString;
+import org.betonquest.betonquest.utils.PlayerConverter;
+import org.bukkit.Bukkit;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.HandlerList;
+import org.bukkit.event.Listener;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This {@link CountingObjective} is completed when a player rides a train for a certain amount of time.
+ * The scripter can specify the train name and the time in seconds the player has to ride the train.
+ * If the train name is not specified, the {@link CountingObjective} will be completed when the player rides any train.
+ */
+public class TrainCartsRideObjective extends CountingObjective implements Listener {
```

Comment:
I am unsure about this implementation in general.
1. the scheduler usage
2. the map that holds Profies (protential leak)
3. how its done in general

Reply:
I'm not sure about it either. If you have the time, we could discuss this in the next few days in a voicechat. Would be more easy